### PR TITLE
feat!: rebalance colors, adhere to style guide & overhaul docs

### PR DESCRIPTION
**Pull request by kjnsn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,25 +17,58 @@
...
-1. [Configuration options](#configuration-options)
-   1. [Window](#window)
-   1. [Pane](#pane)
-   1. [Status Line](#status-line)
-   1. [Using the theme's built-in status modules](#using-the-themes-built-in-status-modules)
-1. [Configuration Examples](#configuration-examples)
-   1. [Config 1](#config-1)
-   1. [Config 2](#config-2)
-   1. [Config 3](#config-3)
+<!--toc:start-->
```
Comment (vdbe) [inline README.md:20]: What is used to generate this toc?

File: catppuccin_tmux.conf [hunk 2]
```diff
@@ -130,21 +130,31 @@ set -wgF pane-border-style "#{E:@catppuccin_pane_border_style}"
...
+  set -gq @catppuccin_window_right_separator " "
 
 %elif "#{==:#{@catppuccin_window_status_style},rounded}"
 
-  set -gq @catppuccin_window_left_separator ""
-  set -gq @catppuccin_window_middle_separator "█"
-  set -gq @catppuccin_window_right_separator " "
+  set -gq @catppuccin_window_left_separator "#[fg=#{@_ctp_status_bg},reverse]#[none]"
+  set -gq @catppuccin_window_middle_separator " "
+  set -gq @catppuccin_window_right_separator "#[fg=#{@_ctp_status_bg},reverse]#[none]"
```
Comment (vdbe) [inline catppuccin_tmux.conf:141]: Not a fan of placing color attributes in this option, I guess it makes the window logic a lot simpler?

**Review by vdbe — COMMENTED** [review]
The default (just running the plugin) and config 3 break with these changes.
- Default
  image
  image
- Config 3
  image
  image

Example config 2 is still broken with these changes, this should be fixed _before_ releasing 2.0.

The example configs should also contain `set -g status-right-length 100` since many people will just copy it an make an issue that it doesn't work.

File: README.md (hunk 1 above)
Comment (kjnsn) [reply README.md:20]: https://github.com/jonschlinkert/markdown-toc

File: catppuccin_tmux.conf (hunk 2 above)
Comment (kjnsn) [reply catppuccin_tmux.conf:141]: Why aren't you a fan? It makes the window logic incredibly simpler

**Comment by kjnsn** [comment]
> The default (just running the plugin) and config 3 break with these changes.
>
> Example config 2 is still broken with these changes, this should be fixed _before_ releasing 2.0.

I am unable to replicate the color issues using alacritty on macos

**Comment by vdbe** [comment]
> > The default (just running the plugin) and config 3 break with these changes.
> > Example config 2 is still broken with these changes, this should be fixed _before_ releasing 2.0.
> 
> I am unable to replicate the color issues using alacritty on macos

I found the issue, it's on my side I loaded the main branch and then switched to this PR to see the diff, without switching it's gone.

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 3]
```diff
@@ -3,6 +3,13 @@ description: Report your bugs with the theme here!
 labels: [bug]
 body:
   - type: checkboxes
+    attributes:
+      label: Something is actually broken, I am not requesting help with my configuration.
+      description: Please open a discussion if you need support.
```
Comment (sgoudham) [inline .github/ISSUE_TEMPLATE/bug.yml:8]: We should probably specify what type of discussion and directly link to it.
You can see my [prior art](https://github.com/catppuccin/jetbrains/blob/270eaff65e91e4914df6840ebb2b6dc7512d5465/.github/ISSUE_TEMPLATE/bug.yml?plain=1#L15) where I figured out that `description` tags allow markdown links.

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 4]
```diff
@@ -3,6 +3,13 @@ description: Report your bugs with the theme here!
 labels: [bug]
 body:
   - type: checkboxes
+    attributes:
+      label: Something is actually broken, I am not requesting help with my configuration.
```
Comment (sgoudham) [inline .github/ISSUE_TEMPLATE/bug.yml:7]: Perhaps the wording should be:
```suggestion
      label: I am not requesting help with my configuration and believe something is genuinely broken.
```
Bit of a minor nit but the reordering feels more natural to read out.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,9 +1,9 @@
 # Contributing
 
-## Adding a module
+## Design
 
-Follow the instructions from [custom](/custom), place the module in the appropriate
-directory and document the module in `README.md`.
+Check out the "[Design Philosphy](./docs/explanation/design.md)" to ensure
+your changes are consistent.
```
Comment (sgoudham) [inline CONTRIBUTING.md:6]: Almost feels like we're missing the last half of the sentence here but maybe just me? 
Personally, I'd reword it to always have the link be "last":

```suggestion
To understand our mindset in developing this plugin and how to ensure your changes align with that mindset, check out the "[Design Philosophy(./docs/explanation/design.md)."
```

What do you think?

(I'd personally appreciate it if the documentation generally followed that as it's also the same as GitHub's documentation style.)

File: README.md [hunk 6]
```diff
@@ -14,91 +15,91 @@
...
+![Mocha Flavor Preview](./assets/mocha.webp)
+
+</details>
+
+## Installation
+
+In order to have the icons displayed correctly please use/update your favorite
+ [nerd font](https://www.nerdfonts.com/font-downloads).
+If you do not have a patched font installed, you can override or remove any
+icon. Check the documentation on the options available.
```
Comment (sgoudham) [inline README.md:52]: Is this referring to a specific document inside `docs/`? If so then I feel like we should link to it.

File: docs/explanation/design.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
+## Design Philosophy
+
+First and foremost, this is a color scheme. Making colors work
+well takes precedence over other kinds of functionality.
```
Comment (sgoudham) [inline docs/explanation/design.md:4]: I like how we are referring to other philosophies but I think I would like to see more content here talking about how _historically_, the plugin kept on growing essentially because no one was there to push back on changes that went against, for example, parts of the UNIX philosophy and how that's made it quite bloated, difficult to maintain, etc. 

Then we can wrap up by saying that we will be trying to align with the philosophies listed below and instead go against what the plugin has historically offered in terms of functionality.

File: docs/reference/configuration.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
+## Configuration Reference
+
+![Default](../../assets/overview.png)
+This is a diagram of how the theme is split between its components.
```
Comment (sgoudham) [inline docs/reference/configuration.md:4]: I know I briefly asked earlier about the usefulness/accuracy of this diagram.

Do you still think it's a net positive to have this diagram even with its inaccuracies? 

I think what I'd prefer is maybe a `[!WARNING]` block at the top saying that this diagram is not up to date and potentially that contributions are welcome if someone wants to take the time to improve it? Suppose it could even be a separate issue if you're interested in that idea.

File: docs/tutorials/01-getting-started.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
+Want to install the color scheme and make tmux pastel? Great! Here's how.
+
+## Step 1: Clone this repository
+
+```bash
+mkdir -p ~/.config/tmux/plugins/catppuccin
+git clone https://github.com/catppuccin/tmux.git ~/.config/tmux/plugins/catppuccin/tmux
```
Comment (sgoudham) [inline docs/tutorials/01-getting-started.md:7]: This should align with the installation in the README where we say to checkout the latest tag (with release please comments making sure that it is up to date)

**Review by sgoudham — CHANGES_REQUESTED** [review]
Amazing work on this PR, really appreciated!

I've left a bunch of comments suggesting changes / asking for your opinion which I think are easy enough to resolve without blocking this PR for too long.

Hope that's okay!

File: docs/reference/configuration.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
+## Configuration Reference
+
+<img src="../../assets/structure.svg" style="background: #eff1f5" />
```
Comment (sgoudham) [inline docs/reference/configuration.md:3]: The CSS in the `style` tag seems to be getting stripped out by GitHub:

image

Probably best to convert to a png?

**Review by sgoudham — CHANGES_REQUESTED** [review]

**Review by sgoudham — APPROVED** [review]
Looks good to me! Thanks again for taking the time to rework this repository.

File: catppuccin_tmux.conf [hunk 11]
```diff
@@ -4,7 +4,7 @@ source -F "#{d:current_file}/themes/catppuccin_#{@catppuccin_flavor}_tmux.conf"
   set -gF @_ctp_status_bg "#{@thm_mantle}"
```
Comment (davidfmatheson) [inline catppuccin_tmux.conf:4]: Should default background be `thm_mantle` or `thm_bg`?

Comment (kjnsn) [reply catppuccin_tmux.conf:4]: The main "background pane" is the terminal window (or tmux pane). The status line is a "secondary pane"

https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md
